aff4/aff4_map.cc

```diff
@@ -99,8 +99,20 @@ AFF4Status AFF4Map::OpenAFF4Map(
             line.erase(line.length()-1, 1);
         }
         AFF4Flusher<AFF4Stream> target;
-        RETURN_IF_ERROR(volumes->GetStream(URN(line), target));
-
+        if((volumes->GetStream(URN(line), target)) != STATUS_OK){
+            // Search in the repository for this target.
+            resolver->logger->info("Attempting to locate resource {} ", line);
```

Comment:
I actually do not think this is a good architectural choice - we are breaking down the encapsulation between classes and assuming a particular context. This makes the code more fragile.

I think it is a mistake to lump volume discovery together with the volume group - volume groups are there to manager lifetimes and they currently do not assume a particular volume implementation (like zip files stored in file backed objects). It is the imager which is aware of the details and it should do the right thing.

IMHO the way this should work is - you try to open the map at which point the map object checks that all the named targets are valid. If any are not known it should return an error status to that effect. The imager should then handle the error if it wants and it makes sense by scanning the volumes near the one we started with. Remember that sometimes the volume may not even be stored on a filesystem and it does not make any sense to try to open it here.